lib/std/math.zig

```diff
@@ -1293,6 +1293,7 @@ pub fn lossyCast(comptime T: type, value: anytype) T {
             }
         },
         .Int => {
+            if (isNan(value)) return 0;
```

Comment:
I'm not convinced that returning 0 is expected behavior when trying to cast NaN to an integer. I think it would be more robust to assert that the value is not NaN if casting to an integer.

Reply:
That might work but I think there could be some cases that could be argued why assert shouldn't be used.

Reply:
> That might work but I think there could be some cases that could be argued why assert shouldn't be used.

Convince me then.

Reply:
If the code is running inside a graphics thread, I don't think immediately crashing would be the best option. The user could do a check to see if the value is valid but by having the check inside the method, it means less operations occurring on the CPU. We could throw an error or return null which I think would be better. However, throwing an error or null wouldn't fit this method imo. `assert` basically would cause an immediate crash.

Reply:
> The user could do a check to see if the value is valid but by having the check inside the method, it means less operations occurring on the CPU. 

I don't follow you here. Its irrelevant whether the branch is inside or outside the function. Also asserting non-NaN will generate better code in release fast/small builds if the user knows the value to be non-NaN without needing a branch.

Returning null might be a fine option but I'm not sure if accidentally passing a NaN to this function is common enough to justify the breaking change.

Reply:
I see, then I'll have it assert.

Reply:
Well, it would be good for users to get a meaningful indication to why their program failed. Looking at the issue, it looks like the message isn't clear.

Reply:
The only other alternative I can think of is to update the documentation. But there's always going to be people who don't read the docs and then gets confused why there's a panic. With an assert, wouldn't there be an error that prints why the error occurred?

Wouldn't having the code and documentation both mention the behavior be the most beneficial?

Reply:
> The only other alternative I can think of is to update the documentation.

The documention needs to be updated in either case.

> But there's always going to be people who don't read the docs and then gets confused why there's a panic. With an assert, wouldn't there be an error that prints why the error occurred?

I don't understand why people would get less confused with an assert.
An asserts is basically just a `panic: reached unreachable code`.
I don't see how that would be better than a panic `@floatToInt` would give.

> Wouldn't having the code and documentation both mention the behavior be the most beneficial?

Does that make a difference? I think someone who doesn't read the documentation would also not read the code.